nexus/serializer.go

```diff
@@ -226,3 +240,22 @@ type compositeSerializer struct {
 var defaultSerializer Serializer = compositeSerializer{
 	serializerChain([]Serializer{nilSerializer{}, byteSliceSerializer{}, jsonSerializer{}}),
 }
+
+type errorMessageFailureConverter struct{}
+
+// ErrorToFailure implements FailureConverter.
+func (e errorMessageFailureConverter) ErrorToFailure(err error) Failure {
+	if err == nil {
+		return Failure{}
+	}
+	return Failure{
+		Message: err.Error(),
+	}
+}
+
+// FailureToError implements FailureConverter.
+func (e errorMessageFailureConverter) FailureToError(f Failure) error {
+	return errors.New(f.Message)
+}
+
+var defaultFailureConverter FailureConverter = errorMessageFailureConverter{}
```

Comment:
People should be allowed to obtain this (and the default serializer) in case they want to delegate to it from their own converters. Would recommend an exported `DefaultFailureConverter() FailureConverter` function (and `DefaultSerializer()` equivalent).

Reply:
Vars are mutable

Reply:
We have plenty of exported vars, why are these ones more important than others to prevent mutations on?

Reply:
We don't have plenty in this repo (I would have discouraged it). The one we do have is for an error which is commonly known in Go not to mutate. We should not expose the default converter as a var though since we access it in a non-thread-safe way. It's harmless to expose as a parameterless function.